seed/src/lib.rs

```diff
@@ -23,6 +23,7 @@ use std::{collections::HashSet, io, net::SocketAddr, path::PathBuf, time::Durati
 
 use futures::stream::StreamExt;
 use thiserror::Error;
+use tokio::sync::mpsc as chan;
```

Comment:
I know it's a lie, but I do not want any tokio in library code unless it is absolutely required. The only exception to this rule are `main` functions, integration tests, and currently `spawn` points.

Reply:
Understandable, but how should we solve this then? I can't use `std::mpsc`, can I?

Reply:
Otherwise I can use a trait, but that seems a bit overkill.

Reply:
`futures::channel::mpsc`